addons/udes_stock_routing/models/stock_picking.py

```diff
@@ -0,0 +1,177 @@
+                    "location_id": configuration.intermediate_location_id,
+                    "location_dest_id": configuration.intermediate_dest_location_id,
+                    "scheduled_date": fields.Datetime.now(),
+                    "priority": original_priority,
+                    "u_from_two_stage_split": True,
+                }
+            )
+            for move in moves:
+                # Duplicate the move to both stages, updating the source, dest, and move orig/dest links accordingly.
+                # move_dest_ids gets cleared on copy()
+                stage_1_move = move.copy(
+                    default={
+                        "location_dest_id": configuration.intermediate_location_id,
+                        "picking_id": stage_1_pick.id,
+                        "picking_type_id": configuration.operation_type_id,
+                        # preserve move_orig_ids, incase this move is not first-in-chain, as the original move will be removed.
+                        "move_orig_ids": [[6, 0, move.move_orig_ids.ids]],
+                    }
+                )
+                _stage_2_move = move.copy(
+                    default={
+                        "location_id": configuration.intermediate_location_id,
+                        "location_dest_id": configuration.intermediate_dest_location_id,
+                        "picking_id": stage_2_pick.id,
+                        "picking_type_id": configuration.operation_type_id,
+                        # preserve move_dest_ids, so if there was a next pick-in-chain then its move will wait for this one.
+                        "move_dest_ids": [[6, 0, move.move_dest_ids.ids]],
+                        # Set the move_orig_ids to the stage_1_move created
+                        "move_orig_ids": [[6, 0, stage_1_move.ids]],
+                    }
+                )
+            # Cancel the original moves which have been split to two stage..
+            # If any 'normal' moves were on this picking, they will be left alone and the picking
+            # will be left available to transfer as normal. Else, the picking will
+            # have no active moves and get cleaned out.
+            moves._action_cancel()
+
+            # Re-reserve the stock for stage 1 pick immediately, so it doesn't get reserved elsewhere.
+            stage_1_pick.action_assign()
+            # Confirm the stage 2 pick.
+            stage_2_pick.action_confirm()
+
+        # Re-assign the pick if there are any active moves left.
+        # This pick doesn't need u_from_two_stage_split flagging as it only contained reserved
+        # stock from 'normal' locations, so another two stage split won't happen when it gets re-reserved.
+        if self.move_lines.filtered(lambda ml: ml.state != "cancel"):
+            self.action_assign()
+
+        # TODO: Consider picks to intermediate location may trigger the putaway push on drop rule, we don't want that to happen
```

Comment:
Needs looking at tomorrow.

Reply:
push on drop rule is not triggered as it applies to the inbound route. so no additional check is required, but in future we _could_ come across a scenario which requires it. We can deal with it as the need arises.